src/client/features/view/components/tooltips/generateContent.js

```diff
@@ -0,0 +1,238 @@
+const h = require('hyperscript');
+const config = require('../../config');
+
+
+//Handle name related metadata fields
+const standardNameHandler = (pair) => makeTooltipItem(pair[1], 'Approved Name: ');
+const displayNameHandler = (pair) => makeTooltipItem(pair[1], 'Display Name: ');
+const nameHandler = (pair, trim) => {
+  //Trim results to first 3 names to avoid overflow
+  let shortArray = pair[1];
+  if (trim) { shortArray = pair[1].slice(0, 3); }
+
+  //Filter out Chemical formulas
+  if (shortArray instanceof Array) shortArray = shortArray.filter(name => (!name.trim().match(/^([^J][0-9BCOHNSOPrIFla@+\-\[\]\(\)\\=#$]{6,})$/ig)));
+
+  //Determine render value
+  let renderValue = h('div.tooltip-value', shortArray.toString());
+  if (!(trim)) {
+    renderValue = makeList(shortArray);
+  }
+
+  return h('div.fake-paragraph', [h('div.field-name', 'Synonyms: '), renderValue]);
+};
+
+//Handle database related fields
+const dataSourceHandler = (pair) => {
+  let source = pair[1].replace('http://pathwaycommons.org/pc2/', '');
+  let link = generateDataSourceLink(source, 'Data Source: ');
+  return h('div.fake-paragraph', link);
+};
+const databaseHandler = (pair, trim) => {
+  //Sort the array by database names
+  let sortedArray = sortByDatabaseId(pair[1]);
+  if (sortedArray.length < 1) { return; }
+  return h('div.fake-paragraph',
+    [
+      h('div.field-name', 'Database References:'),
+      h('div.wrap-text', h('ul.db-list', sortedArray.map(item => generateIdList(item, trim), this)))
+    ]);
+};
+
+//Handle type related fields
+const typeHandler = (pair, trim, key) => {
+  if (!(trim)) {
```

Comment:
Same problems here.  Don't use a trim flag here and always ensure you return html.

Reply:
Removed trim parameter from all handler functions and created separate functions to handle cases involving trimmed values.